worker/worker.go

```diff
@@ -936,5 +950,35 @@ func New(masterKey [32]byte, b Bus, sessionReconectTimeout, sessionTTL time.Dura
 		"GET    /objects/*key": w.objectsKeyHandlerGET,
 		"PUT    /objects/*key": w.objectsKeyHandlerPUT,
 		"DELETE /objects/*key": w.objectsKeyHandlerDELETE,
-	}), nil
+	}), cleanup, nil
+}
+
+func (w *worker) recordInteractions(interactions []hostdb.Interaction) {
+	w.interactionsMu.Lock()
+	defer w.interactionsMu.Unlock()
+
+	// Append interactions to buffer.
+	w.interactions = append(w.interactions, interactions...)
+
+	// If a thread was scheduled to flush the buffer we are done.
+	if w.interactionsFlushTimer != nil {
+		return
+	}
+	// Otherwise we schedule a flush.
+	w.interactionsFlushTimer = time.AfterFunc(w.interactionsFlushInterval, func() {
```

Comment:
`time.AfterFunc(w.interactionsFlushInterval, w.flushInteractions)` and lock in `flushInteractions`?